### ISSUE TITLE
[Issue]: Metadata retrieval from TheMovieDB stalls, and then prevents library scan from completing

### ISSUE THREAD

**Issue by aoberoi** [jellyfin/jellyfin#10584]
### Please describe your bug

I ran into an issue with the initial scan of my media files in a library with content type "Shows". The scan initializes but then there's no feedback whatsoever, many minutes later I saw one episode, unassociated to a show, that had no metadata. The remaining files (a total of 4) never appeared. The logs didn't give any clear indication of what was happening. After a painful troubleshooting process that took many hours, on a whim, I decided to disable TheMovieDB as a metadata source. This allowed the scan to complete, for the show to appear, its corresponding seasons, and episodes.

Based on this experience, I think there are several changes we could make to Jellyfin, to help users (especially new ones) avoid the roadblock I faced. Here are some ideas: 

* Ability to skip slow or failed metadata sources during scan process. Essentially robustness in the face of a misbehaving or slow metadata API.
* More visibility into what is happening during the scan process, especially the metadata retrieval, for debugging.
    * Optionally, making this visible in the UI. I can imagine a place to cancel (or even rollback) the job(s) in progress.
* An investigation into why retrieval from TheMovieDB continued to fail, and whether we can fix this on the client side (as in Jellyfin server, as a client of TheMovieDB's API).

### Jellyfin Version

10.8.z

### if other:

10.8.12.0

### Environment

```markdown
- OS: Raspberry Pi OS (Debian)
- Linux Kernel: 6.1.0
- Virtualization:
- Clients: Web
- Browser: Chromium
- FFmpeg Version: 
- Playback Method:
- Hardware Acceleration:
- GPU Model:
- Plugins:
- Reverse Proxy:
- Base URL:
- Networking:
- Storage: USB attached SSD
```

### Jellyfin logs

```shell
Nov 14 11:18:09 famshare jellyfin[18562]: [11:18:09] [DBG] Refreshing Folder /var/lib/jellyfin/root because item.RequiresRefresh() returned true
Nov 14 11:18:09 famshare jellyfin[18562]: [11:18:09] [DBG] Running FolderImageProvider for /var/lib/jellyfin/root
Nov 14 11:18:10 famshare jellyfin[18562]: [11:18:10] [DBG] Running FolderImageProvider for /media/FamShare/Media/Shows
Nov 14 11:18:10 famshare jellyfin[18562]: [11:18:10] [DBG] Running CollectionFolderImageProvider for /var/lib/jellyfin/root/default/Shows
Nov 14 11:18:10 famshare jellyfin[18562]: [11:18:10] [DBG] CollectionFolderImageProvider reports change to /var/lib/jellyfin/root/default/Shows
Nov 14 11:18:10 famshare jellyfin[18562]: [11:18:10] [DBG] Running CollectionFolderImageProvider for /var/lib/jellyfin/root/default/Shows
Nov 14 11:18:10 famshare jellyfin[18562]: [11:18:10] [INF] Queuing task RefreshMediaLibraryTask
Nov 14 11:18:10 famshare jellyfin[18562]: [11:18:10] [INF] Executing Scan Media Library
Nov 14 11:18:10 famshare jellyfin[18562]: [11:18:10] [WRN] Slow HTTP Response from http://192.168.0.147:8096/Library/VirtualFolders?collectionType=tvshows&refreshLibrary=true&name=Shows to 192.168.0.119 in 0:00:01.5400845 with Status Code 204
Nov 14 11:18:10 famshare jellyfin[18562]: [11:18:10] [INF] Validating media library
Nov 14 11:18:10 famshare jellyfin[18562]: [11:18:10] [DBG] Refreshing Folder /var/lib/jellyfin/root because item.RequiresRefresh() returned true
Nov 14 11:18:10 famshare jellyfin[18562]: [11:18:10] [DBG] Running FolderImageProvider for /var/lib/jellyfin/root
Nov 14 11:18:10 famshare jellyfin[18562]: [11:18:10] [DBG] CollectionFolderImageProvider reports change to /var/lib/jellyfin/root/default/Shows
Nov 14 11:18:10 famshare jellyfin[18562]: [11:18:10] [DBG] Running CollectionFolderImageProvider for /var/lib/jellyfin/root/default/Shows
Nov 14 11:18:10 famshare jellyfin[18562]: [11:18:10] [DBG] CollectionFolderImageProvider reports change to /var/lib/jellyfin/root/default/Shows
Nov 14 11:18:10 famshare jellyfin[18562]: [11:18:10] [DBG] Running CollectionFolderImageProvider for /var/lib/jellyfin/root/default/Shows
Nov 14 11:18:10 famshare jellyfin[18562]: [11:18:10] [DBG] OnRefreshStart f27caa37e5142225cceded48f6553502
Nov 14 11:18:10 famshare jellyfin[18562]: [11:18:10] [DBG] OnRefreshProgress f27caa37e5142225cceded48f6553502 5
Nov 14 11:18:10 famshare jellyfin[18562]: [11:18:10] [DBG] OnRefreshProgress f27caa37e5142225cceded48f6553502 10
Nov 14 11:18:10 famshare jellyfin[18562]: [11:18:10] [DBG] OnRefreshStart 1071671e7bffa0532e930debee501d2e
Nov 14 11:18:10 famshare jellyfin[18562]: [11:18:10] [DBG] OnRefreshStart 5a01eb9c6f7c38219b5b6815bebdebeb
Nov 14 11:18:10 famshare jellyfin[18562]: [11:18:10] [DBG] OnRefreshProgress f27caa37e5142225cceded48f6553502 11
Nov 14 11:18:10 famshare jellyfin[18562]: [11:18:10] [DBG] OnRefreshProgress 1071671e7bffa0532e930debee501d2e 5
Nov 14 11:18:10 famshare jellyfin[18562]: [11:18:10] [DBG] OnRefreshProgress f27caa37e5142225cceded48f6553502 12
Nov 14 11:18:10 famshare jellyfin[18562]: [11:18:10] [DBG] OnRefreshProgress 1071671e7bffa0532e930debee501d2e 10
Nov 14 11:18:10 famshare jellyfin[18562]: [11:18:10] [DBG] OnRefreshComplete 1071671e7bffa0532e930debee501d2e
Nov 14 11:18:10 famshare jellyfin[18562]: [11:18:10] [DBG] AuthenticationScheme: CustomAuthentication was successfully authenticated.
Nov 14 11:18:10 famshare jellyfin[18562]: [11:18:10] [DBG] AuthenticationScheme: CustomAuthentication was successfully authenticated.
Nov 14 11:18:10 famshare jellyfin[18562]: [11:18:10] [DBG] Getting topLibraryFolders
Nov 14 11:18:10 famshare jellyfin[18562]: [11:18:10] [DBG] Getting refreshQueue
Nov 14 11:18:10 famshare jellyfin[18562]: [11:18:10] [DBG] OnRefreshProgress f27caa37e5142225cceded48f6553502 30
Nov 14 11:18:10 famshare jellyfin[18562]: [11:18:10] [DBG] OnRefreshProgress f27caa37e5142225cceded48f6553502 31
Nov 14 11:18:10 famshare jellyfin[18562]: [11:18:10] [DBG] OnRefreshProgress 5a01eb9c6f7c38219b5b6815bebdebeb 5
Nov 14 11:18:10 famshare jellyfin[18562]: [11:18:10] [DBG] OnRefreshProgress f27caa37e5142225cceded48f6553502 32
Nov 14 11:18:10 famshare jellyfin[18562]: [11:18:10] [DBG] OnRefreshProgress 5a01eb9c6f7c38219b5b6815bebdebeb 10
Nov 14 11:18:10 famshare jellyfin[18562]: [11:18:10] [DBG] OnRefreshStart 881eb0862bd3f5e47cc08d10de6c5b54
Nov 14 11:18:10 famshare jellyfin[18562]: [11:18:10] [DBG] OnRefreshProgress f27caa37e5142225cceded48f6553502 32.400000000000006
Nov 14 11:18:10 famshare jellyfin[18562]: [11:18:10] [DBG] OnRefreshProgress 5a01eb9c6f7c38219b5b6815bebdebeb 12
Nov 14 11:18:10 famshare jellyfin[18562]: [11:18:10] [DBG] OnRefreshProgress 881eb0862bd3f5e47cc08d10de6c5b54 5
Nov 14 11:18:10 famshare jellyfin[18562]: [11:18:10] [DBG] OnRefreshProgress f27caa37e5142225cceded48f6553502 32.8
Nov 14 11:18:10 famshare jellyfin[18562]: [11:18:10] [DBG] OnRefreshProgress 5a01eb9c6f7c38219b5b6815bebdebeb 14
Nov 14 11:18:10 famshare jellyfin[18562]: [11:18:10] [DBG] OnRefreshProgress 881eb0862bd3f5e47cc08d10de6c5b54 10
Nov 14 11:18:10 famshare jellyfin[18562]: [11:18:10] [DBG] OnRefreshStart 035d28356141958313cba63b8b8e23a2
Nov 14 11:18:10 famshare jellyfin[18562]: [11:18:10] [DBG] OnRefreshStart 477527eadef19d4ad03a062e2f0051c6
Nov 14 11:18:10 famshare jellyfin[18562]: [11:18:10] [DBG] OnRefreshStart 619639572c5422e38f272e711129247f
Nov 14 11:18:11 famshare jellyfin[18562]: [11:18:11] [DBG] OnRefreshProgress 881eb0862bd3f5e47cc08d10de6c5b54 11.333333333333334
Nov 14 11:18:11 famshare jellyfin[18562]: [11:18:11] [DBG] OnRefreshProgress 5a01eb9c6f7c38219b5b6815bebdebeb 14.4
Nov 14 11:18:11 famshare jellyfin[18562]: [11:18:11] [DBG] OnRefreshProgress f27caa37e5142225cceded48f6553502 33
Nov 14 11:18:11 famshare jellyfin[18562]: [11:18:11] [DBG] OnRefreshProgress 881eb0862bd3f5e47cc08d10de6c5b54 10.666666666666666
Nov 14 11:18:11 famshare jellyfin[18562]: [11:18:11] [DBG] OnRefreshProgress 477527eadef19d4ad03a062e2f0051c6 5
Nov 14 11:18:11 famshare jellyfin[18562]: [11:18:11] [DBG] OnRefreshProgress 619639572c5422e38f272e711129247f 5
Nov 14 11:18:11 famshare jellyfin[18562]: [11:18:11] [DBG] OnRefreshProgress 5a01eb9c6f7c38219b5b6815bebdebeb 14.8
Nov 14 11:18:11 famshare jellyfin[18562]: [11:18:11] [DBG] OnRefreshProgress 881eb0862bd3f5e47cc08d10de6c5b54 12
Nov 14 11:18:11 famshare jellyfin[18562]: [11:18:11] [DBG] OnRefreshProgress 035d28356141958313cba63b8b8e23a2 5
Nov 14 11:18:11 famshare jellyfin[18562]: [11:18:11] [DBG] OnRefreshProgress 5a01eb9c6f7c38219b5b6815bebdebeb 15.2
Nov 14 11:18:11 famshare jellyfin[18562]: [11:18:11] [DBG] OnRefreshProgress 881eb0862bd3f5e47cc08d10de6c5b54 12.666666666666666
Nov 14 11:18:11 famshare jellyfin[18562]: [11:18:11] [DBG] OnRefreshProgress 619639572c5422e38f272e711129247f 10
Nov 14 11:18:11 famshare jellyfin[18562]: [11:18:11] [DBG] OnRefreshComplete 619639572c5422e38f272e711129247f
Nov 14 11:18:11 famshare jellyfin[18562]: [11:18:11] [DBG] OnRefreshProgress f27caa37e5142225cceded48f6553502 34
Nov 14 11:18:11 famshare jellyfin[18562]: [11:18:11] [DBG] OnRefreshProgress 5a01eb9c6f7c38219b5b6815bebdebeb 20
Nov 14 11:18:11 famshare jellyfin[18562]: [11:18:11] [DBG] OnRefreshProgress 881eb0862bd3f5e47cc08d10de6c5b54 24.666666666666664
Nov 14 11:18:11 famshare jellyfin[18562]: [11:18:11] [DBG] OnRefreshProgress 881eb0862bd3f5e47cc08d10de6c5b54 25.333333333333336
Nov 14 11:18:11 famshare jellyfin[18562]: [11:18:11] [DBG] OnRefreshProgress 477527eadef19d4ad03a062e2f0051c6 10
Nov 14 11:18:11 famshare jellyfin[18562]: [11:18:11] [DBG] OnRefreshComplete 477527eadef19d4ad03a062e2f0051c6
Nov 14 11:18:11 famshare jellyfin[18562]: [11:18:11] [DBG] OnRefreshProgress f27caa37e5142225cceded48f6553502 35
Nov 14 11:18:11 famshare jellyfin[18562]: [11:18:11] [DBG] OnRefreshProgress 5a01eb9c6f7c38219b5b6815bebdebeb 24.8
Nov 14 11:18:11 famshare jellyfin[18562]: [11:18:11] [DBG] OnRefreshProgress 881eb0862bd3f5e47cc08d10de6c5b54 37.33333333333333
Nov 14 11:18:11 famshare jellyfin[18562]: [11:18:11] [DBG] OnRefreshProgress 5a01eb9c6f7c38219b5b6815bebdebeb 25.2
Nov 14 11:18:11 famshare jellyfin[18562]: [11:18:11] [DBG] OnRefreshProgress 881eb0862bd3f5e47cc08d10de6c5b54 38
Nov 14 11:18:11 famshare jellyfin[18562]: [11:18:11] [DBG] OnRefreshProgress 035d28356141958313cba63b8b8e23a2 10
Nov 14 11:18:11 famshare jellyfin[18562]: [11:18:11] [DBG] OnRefreshComplete 035d28356141958313cba63b8b8e23a2
Nov 14 11:18:11 famshare jellyfin[18562]: [11:18:11] [DBG] OnRefreshProgress f27caa37e5142225cceded48f6553502 36
Nov 14 11:18:11 famshare jellyfin[18562]: [11:18:11] [DBG] OnRefreshProgress 5a01eb9c6f7c38219b5b6815bebdebeb 30
Nov 14 11:18:11 famshare jellyfin[18562]: [11:18:11] [DBG] OnRefreshProgress 881eb0862bd3f5e47cc08d10de6c5b54 50
Nov 14 11:18:11 famshare jellyfin[18562]: [11:18:11] [DBG] OnRefreshComplete 881eb0862bd3f5e47cc08d10de6c5b54
Nov 14 11:18:11 famshare jellyfin[18562]: [11:18:11] [DBG] OnRefreshProgress f27caa37e5142225cceded48f6553502 40
Nov 14 11:18:11 famshare jellyfin[18562]: [11:18:11] [DBG] OnRefreshProgress 5a01eb9c6f7c38219b5b6815bebdebeb 50
Nov 14 11:18:11 famshare jellyfin[18562]: [11:18:11] [DBG] OnRefreshComplete 5a01eb9c6f7c38219b5b6815bebdebeb
Nov 14 11:18:11 famshare jellyfin[18562]: [11:18:11] [DBG] OnRefreshProgress f27caa37e5142225cceded48f6553502 50
Nov 14 11:18:11 famshare jellyfin[18562]: [11:18:11] [DBG] OnRefreshProgress f27caa37e5142225cceded48f6553502 50
Nov 14 11:18:11 famshare jellyfin[18562]: [11:18:11] [DBG] OnRefreshProgress f27caa37e5142225cceded48f6553502 75
Nov 14 11:18:11 famshare jellyfin[18562]: [11:18:11] [DBG] GetItemList query time (slow): 11.9771ms. Query: select type,data,StartDate,EndDate,ChannelId,IsMovie,IsSeries,EpisodeTitle,IsRepeat,CommunityRating,CustomRating,IndexNumber,IsLocked,PreferredMetadataLanguage,PreferredMetadataCountryCode,Width,Height,DateLastRefreshed,Name,Path,PremiereDate,Overview,ParentIndexNumber,ProductionYear,OfficialRating,ForcedSortName,RunTimeTicks,Size,DateCreated,DateModified,guid,Genres,ParentId,Audio,ExternalServiceId,IsInMixedFolder,DateLastSaved,LockedFields,Studios,Tags,TrailerTypes,OriginalTitle,PrimaryVersionId,DateLastMediaAdded,Album,CriticRating,IsVirtualItem,SeriesName,SeasonName,SeasonId,SeriesId,PresentationUniqueKey,InheritedParentalRatingValue,ExternalSeriesId,Tagline,ProviderIds,Images,ProductionLocations,ExtraIds,TotalBitrate,ExtraType,Artists,AlbumArtists,ExternalId,SeriesPresentationUniqueKey,ShowId,OwnerId from TypedBaseItems A where ParentId=@ParentId
Nov 14 11:18:11 famshare jellyfin[18562]: [11:18:11] [DBG] Running SeriesNfoProvider for /media/FamShare/Media/Shows/Survivor
Nov 14 11:18:11 famshare jellyfin[18562]: [11:18:11] [DBG] SeriesNfoProvider returned no metadata for /media/FamShare/Media/Shows/Survivor
Nov 14 11:18:11 famshare jellyfin[18562]: [11:18:11] [DBG] Running TvdbSeriesProvider for /media/FamShare/Media/Shows/Survivor
Nov 14 11:18:11 famshare jellyfin[18562]: [11:18:11] [DBG] TvdbSearch: Finding id for item: Survivor (null)
Nov 14 11:18:16 famshare jellyfin[18562]: [11:18:16] [DBG] Running TmdbSeriesProvider for /media/FamShare/Media/Shows/Survivor
Nov 14 11:18:41 famshare jellyfin[18562]: [11:18:41] [DBG] GetItemList query time (slow): 19.431ms. Query: select type,data,StartDate,EndDate,ChannelId,IsMovie,IsSeries,EpisodeTitle,IsRepeat,CommunityRating,CustomRating,IndexNumber,IsLocked,PreferredMetadataLanguage,PreferredMetadataCountryCode,Width,Height,DateLastRefreshed,Name,Path,PremiereDate,Overview,ParentIndexNumber,ProductionYear,OfficialRating,ForcedSortName,RunTimeTicks,Size,DateCreated,DateModified,guid,Genres,ParentId,Audio,ExternalServiceId,IsInMixedFolder,DateLastSaved,LockedFields,Studios,Tags,TrailerTypes,OriginalTitle,PrimaryVersionId,DateLastMediaAdded,Album,CriticRating,IsVirtualItem,SeriesName,SeasonName,SeasonId,SeriesId,PresentationUniqueKey,InheritedParentalRatingValue,ExternalSeriesId,Tagline,ProviderIds,Images,ProductionLocations,ExtraIds,TotalBitrate,ExtraType,Artists,AlbumArtists,ExternalId,SeriesPresentationUniqueKey,ShowId,OwnerId from TypedBaseItems A where ParentId=@ParentId
Nov 14 11:18:41 famshare jellyfin[18562]: [11:18:41] [DBG] AuthenticationScheme: CustomAuthentication was successfully authenticated.
Nov 14 11:18:41 famshare jellyfin[18562]: [11:18:41] [DBG] AuthenticationScheme: CustomAuthentication was successfully authenticated.
Nov 14 11:18:41 famshare jellyfin[18562]: [11:18:41] [DBG] GetItemList query time (slow): 13.2561ms. Query: select type,data,StartDate,EndDate,ChannelId,IsMovie,IsSeries,EpisodeTitle,IsRepeat,CommunityRating,IndexNumber,Width,Height,Name,Path,PremiereDate,ParentIndexNumber,ProductionYear,OfficialRating,RunTimeTicks,Size,DateModified,guid,ParentId,Audio,ExternalServiceId,IsInMixedFolder,Tags,TrailerTypes,PrimaryVersionId,Album,CriticRating,IsVirtualItem,SeriesName,SeasonName,SeasonId,SeriesId,ProviderIds,Images,TotalBitrate,ExtraType,Artists,AlbumArtists,ExternalId,ShowId,OwnerId from TypedBaseItems A where IsFolder=@IsFolder AND MediaType in ('Audio','Video') AND (Guid = @IncludeId0 OR Guid = @IncludeId1 OR Guid = @IncludeId2 OR Guid = @IncludeId3 OR Guid = @IncludeId4 OR Guid = @IncludeId5 OR Guid = @IncludeId6 OR Guid = @IncludeId7 OR Guid = @IncludeId8 OR Guid = @IncludeId9) Group by PresentationUniqueKey ORDER BY DateCreated DESC LIMIT 3
Nov 14 11:19:56 famshare jellyfin[18562]: [11:19:56] [DBG] OnRefreshProgress f27caa37e5142225cceded48f6553502 100
Nov 14 11:19:56 famshare jellyfin[18562]: [11:19:56] [DBG] OnRefreshComplete f27caa37e5142225cceded48f6553502
Nov 14 11:19:56 famshare jellyfin[18562]: [11:19:56] [DBG] Running post-scan task SplashscreenPostScanTask
Nov 14 11:19:56 famshare jellyfin[18562]: [11:19:56] [DBG] No thumb images found. Using backdrops to generate splashscreen
Nov 14 11:19:56 famshare jellyfin[18562]: [11:19:56] [ERR] Error running post-scan task
Nov 14 11:19:56 famshare jellyfin[18562]: System.ArgumentException: Not enough valid pictures provided to create a splashscreen!
Nov 14 11:19:56 famshare jellyfin[18562]:    at Jellyfin.Drawing.Skia.SplashscreenBuilder.GenerateCollage(IReadOnlyList`1 posters, IReadOnlyList`1 backdrops)
Nov 14 11:19:56 famshare jellyfin[18562]:    at Jellyfin.Drawing.Skia.SplashscreenBuilder.GenerateSplash(IReadOnlyList`1 posters, IReadOnlyList`1 backdrops, String outputPath)
Nov 14 11:19:56 famshare jellyfin[18562]:    at Jellyfin.Drawing.Skia.SkiaEncoder.CreateSplashscreen(IReadOnlyList`1 posters, IReadOnlyList`1 backdrops)
Nov 14 11:19:56 famshare jellyfin[18562]:    at Emby.Server.Implementations.Library.SplashscreenPostScanTask.Run(IProgress`1 progress, CancellationToken cancellationToken)
Nov 14 11:19:56 famshare jellyfin[18562]:    at Emby.Server.Implementations.Library.LibraryManager.RunPostScanTasks(IProgress`1 progress, CancellationToken cancellationToken)
Nov 14 11:19:56 famshare jellyfin[18562]: [11:19:56] [DBG] Running post-scan task ArtistsPostScanTask
Nov 14 11:19:56 famshare jellyfin[18562]: [11:19:56] [DBG] Running post-scan task CollectionPostScanTask
Nov 14 11:19:56 famshare jellyfin[18562]: [11:19:56] [DBG] Running post-scan task GenresPostScanTask
Nov 14 11:19:56 famshare jellyfin[18562]: [11:19:56] [DBG] Running post-scan task MusicGenresPostScanTask
Nov 14 11:19:56 famshare jellyfin[18562]: [11:19:56] [DBG] Running post-scan task StudiosPostScanTask
Nov 14 11:19:56 famshare jellyfin[18562]: [11:19:56] [DBG] Running post-scan task CleanDatabaseScheduledTask
Nov 14 11:19:56 famshare jellyfin[18562]: [11:19:56] [DBG] Cleaning 0 items with dead parent links
Nov 14 11:19:56 famshare jellyfin[18562]: [11:19:56] [INF] Scan Media Library Completed after 1 minute(s) and 46 seconds
Nov 14 11:19:56 famshare jellyfin[18562]: [11:19:56] [INF] ExecuteQueuedTasks
Nov 14 11:19:56 famshare jellyfin[18562]: [11:19:56] [INF] Watching directory /media/FamShare/Media/Shows
```

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by aoberoi** [comment]
Troubleshooting done in the matrix chat, most of it in this thread: https://matrix.to/#/!YjAUNWwLVbCthyFrkz:bonifacelabs.ca/$TuzlCSN_g20S79pm4LhZQzbo47O7SgJDKbejiIx6Kl8?via=matrix.org&via=t2bot.io&via=bitwrk.de

**Comment by cchristchurch** [comment]
:+1: This is by far the biggest issue I'm experiencing with Jellyfin.

I've spent hours and hours trying to get metadata retrieval to work correctly since setting up my server 3 months ago. The main problem is that even when set to DEBUG log level, Jellyfin logs close to zero useful information about the metadata retrieval process. 

Adding a few logs here would probably be a fairly easy way to *massively* improve the admin experience. I'm happy to help but with absolutely no knowledge of .NET it's likely I'd do more harm than good. I think we'd need the following extra logs (in DEBUG level):
- Log the full URL for every HTTP GET request sent by the metadata retriever
- If possible log the IP that the request is being sent out to (if that's too low level it would be nice to know if the library is using ipv6 or ipv4 at least)
- In case of timeout, log it
- In case of error response, log as much info as possible
- In case of success response, log the content of the response (I agree that may cause fairly verbose logs for large library scans but after all isn't that what DEBUG log level is for?)

Some examples of problems that this would greatly help with:
- Initially I wasn't aware of the required naming conventions and the identification of some movies would fail. There is currently no information in the logs that make this different from, say, a network failure. After I renamed the offending movies the scans would work normally.
- Then a few weeks ago I broke metadata retrieval due to a firewall rule that was blocking some outbound ipv6 connections from the jellyfin box. Because I thought this was a repeat of the previous problem I spent a few hours fiddling with file names before realizing it was a different problem. A simple log with the HTTP GET request and the corresponding timeout would have greatly helped me.
- As of today metadata retrieval seems broken again on my server and I have not yet been able to figure out why. The movies seem to be correctly assigned their TMDB IDs, but no metadata is being retrieved. I tried executing `curl --request GET --url "https://api.themoviedb.org/3/movie/11?api_key=JELLYFIN_API_KEY"` on the box and it seems to work fine so it does not seem to be a connectivity issue this time. I suspect the retriever is making more requests with different parameters and maybe some of those are temporarily failing, but again that's very hard to say without logs.

**Comment by cchristchurch** [comment]
I suspect [this](https://github.com/jellyfin/TMDbLib/blob/d2893c9d8de4d19cb2a2c178faa56a8b0a1416f9/TMDbLib/Client/TMDbClient.cs) could be where most of the relevant logs could be added.

That would only cover TheMoviedDB though and possibly other retrievers would need the same

**Comment by cchristchurch** [comment]
Also I can confirm I currently see the same behavior as originally reported here by @aoberoi : whatever error is happening is causing the scan to completely stall or fail and then no movies get added to the database ... so *no new movie can get added to my library* (even without metadata, even if I put the metadata manually in the jellyfin UI).

Given the impact of the bug (essentially making jellyfin completely unusable) it seems really important to at least log something. In the long term I also agree it would be nice to actually *handle* the error better, and maybe just not make my library unusable if TMDB stops returning data, but I suspect that is significantly more work, so possibly a quicker log-only fix would be preferrable before we start thinking how these rerrors could be handled better and / or exposed through the UI.

**Comment by cchristchurch** [comment]
Final update for the day, I finally figured out what my new issue was (or rather, how to fix it): **If I disable the second metadata retriever (The Open Movie Database), things work perfectly again.**

I disabled it both as a Metadata Retriever and as an Image Fetcher. Now the metadata and images are successfully retrieved from TheMovieDB and my jellyfin database is updated correctly again when I add a new movie.

So to summarize, quite a few different problems here:
* Why is the open movie database retriever no longer working? (I don't really care that much about this TBH, I've never used it as a primary retriever)
* Why does it completely block the update process and the addition of new movies to my db, in spite of the fact it's my secondary retriever, and most if not all metadata is successfully retrieved from the primary retriever (TheMovieDB)? That issue looks similar to the one reported by @aoberoi ; as suspected in https://github.com/jellyfin/jellyfin/issues/10584#issuecomment-1841333845 it confirms this issue is *not* specific to TheMovieDB. It also looks like this issue can also happen if the offending retriever is not the primary and only used as a fallback.
* Why do I not see any logs for this? :smile: at the risk of sounding like a broken record I reallly think that's the most urgent issue to fix here. None of these issues would really be painful if I could just look at the source of the problem in the logs and disable the offending plugin in 5 minutes.

**Comment by cchristchurch** [comment]
This is potentially making jellyfin completely unusable by default for thousands of users.
Adding logs to help troubleshooting seems easy.
I really don't think this should be closed unless the project is no longer maintained.

**Comment by Bond-009** [comment]
Please check with the latest release, 10.9.2 at this point in time and report back.

**Comment by cchristchurch** [comment]
Thanks for looking into this @Bond-009 

I've just upgraded to `10.9.2+deb12` and I attempted to reproduce the issue:
* I switched to DEBUG logging mode
* I re-enabled The Open Movie Database as a second metadata and image retriever
* I triggered a full metadata refresh on my movie library (using the "Replace All Metadata" options)

During the refresh, I see the following log pattern for every movie:
```
[2024-05-19 00:45:48.405 +01:00] [DBG] Running "ProbeProvider" for "/path/to/library/SharedMovies/Some Movie Name (2011)/Some Movie Name (2011) - MULTi 1080p x264.mkv"
[2024-05-19 00:45:48.405 +01:00] [INF] Starting "/usr/lib/jellyfin-ffmpeg/ffprobe" with args "-analyzeduration 200M -probesize 1G -i file:\"/path/to/library/SharedMovies/Some Movie Name (2011)/Some Movie Name (2011) - MULTi 1080p x264.mkv\" -threads 0 -v warning -print_format json -show_streams -show_chapters -show_format"
[2024-05-19 00:45:48.515 +01:00] [DBG] Running "TmdbMovieProvider" for "/path/to/library/SharedMovies/Some Movie Name (2011)/Some Movie Name (2011) - MULTi 1080p x264.mkv"
[2024-05-19 00:45:48.595 +01:00] [DBG] Getting image size for item "Movie" "/var/lib/jellyfin/metadata/library/27/27e2b0372776fcb494575fd65877b465/landscape.jpg"
[2024-05-19 00:45:48.612 +01:00] [DBG] Running "OmdbItemProvider" for "/path/to/library/SharedMovies/Some Movie Name (2011)/Some Movie Name (2011) - MULTi 1080p x264.mkv"
[2024-05-19 00:45:48.627 +01:00] [DBG] Running "MovieNfoProvider" for "/path/to/library/SharedMovies/Some Movie Name (2011)/Some Movie Name (2011) - MULTi 1080p x264.mkv"
[2024-05-19 00:45:48.629 +01:00] [DBG] "MovieNfoProvider" returned no metadata for "/path/to/library/SharedMovies/Some Movie Name (2011)/Some Movie Name (2011) - MULTi 1080p x264.mkv"
[2024-05-19 00:45:48.629 +01:00] [DBG] Running "TrickplayProvider" for "/path/to/library/SharedMovies/Some Movie Name (2011)/Some Movie Name (2011) - MULTi 1080p x264.mkv"
[2024-05-19 00:45:48.629 +01:00] [DBG] Trickplay refresh for 00917e59-a7db-5129-e112-f938cb6a465a (replace existing: False)
[2024-05-19 00:45:48.630 +01:00] [DBG] Running "TmdbMovieImageProvider" for "/path/to/library/SharedMovies/Some Movie Name (2011)/Some Movie Name (2011) - MULTi 1080p x264.mkv"
[2024-05-19 00:45:48.654 +01:00] [DBG] Saving image to "/var/lib/jellyfin/metadata/library/00/00917e59a7db5129e112f164cb6a3f89/landscape.jpg"
```

From the above, I see which retriever gets called for which movie but the logs still seem to be missing information that can be useful to troubleshoot a problem (see [previous post](https://github.com/jellyfin/jellyfin/issues/10584#issuecomment-1864682276) for suggestions on potentially useful information to log).

The scan completed and the movies are appearing correctly, however without more information in the logs it's a bit unclear exactly what has been fixed regarding the issues listed [here](https://github.com/jellyfin/jellyfin/issues/10584#issuecomment-1864896901):
* Maybe The Open Movie Database is still failing but Jellyfin is handling the failure gracefully, and the library scan completes normally?
* Maybe the Open Movie Database retriever was fixed and is no longer failing?
* Maybe something was fixed on the server side at https://www.omdbapi.com/ which causes data retrieval to work where it would previously fail?

In short, it's unclear which issue, if any, have been fixed on the jellyfin side. I think it's worth keeping this open, if only to track the addition of a couple of debug logs.
